File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/tierradelfuego/domain/TierraDelFuegoObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.tierradelfuego.domain;

public abstract class TierraDelFuegoObject {
    public abstract String objAsString();
    public static TierraDelFuegoObject objFromString(String str) {
        switch (str) {
            case "marker":
                return new TierraDelFuegoMarkerObject();
            case "tree":
                return new TierraDelFuegoTreeObject();
            default:
                return new TierraDelFuegoEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/tierradelfuego/TierraDelFuegoObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.tierradelfuego

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

enum class TierraDelFuegoObject {
    Empty, Forbidden, Hint, Marker, Water
}

class TierraDelFuegoGameMove(val p: Position, var obj: TierraDelFuegoObject = TierraDelFuegoObject.Empty)
